packages/trace-viewer/src/ui/teleSuiteUpdater.ts

```diff
@@ -145,3 +149,36 @@ export class TeleSuiteUpdater {
     };
   }
 }
+
+class TestResultsSnapshot {
+  private _testIdToResults = new Map<string, TeleTestResult[]>();
+
+  static saveResults(rootSuite: TeleSuite) {
+    const snapshot = new TestResultsSnapshot();
+    snapshot._saveForSuite(rootSuite);
+    return snapshot;
+  }
+
+  private _saveForSuite(suite: TeleSuite) {
+    for (const entry of suite.entries()) {
+      if (entry.type === 'test') {
+        if (entry.results.length)
+          this._testIdToResults.set(entry.id, entry.results);
+      } else {
+        this._saveForSuite(entry);
+      }
+    }
+  }
+
+  restore(suite: TeleSuite) {
+    for (const entry of suite.entries()) {
+      if (entry.type === 'test') {
+        const results = this._testIdToResults.get(entry.id);
+        if (results)
+          entry.results = results;
```

Comment:
This does not update `TeleTestCase._resultsMap`. Perhaps introduce something like `_addResult()`?